src/transport/raw/BLE.cpp

```diff
@@ -0,0 +1,210 @@
+    BleLayer * bleLayer = param.GetBleLayer();
+
+    VerifyOrExit(mState == State::kNotReady, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(bleLayer != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    mBleLayer                           = bleLayer;
+    mBleLayer->mBleTransport            = this;
+    mBleLayer->OnChipBleConnectReceived = nullptr;
+
+    mState = State::kInitialized;
+
+    SuccessOrExit(err);
+
+exit:
+    return err;
+}
+
+CHIP_ERROR BLEBase::SetEndPoint(Ble::BLEEndPoint * endPoint)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    VerifyOrExit(endPoint->mState == BLEEndPoint::kState_Connected, err = CHIP_ERROR_INVALID_ARGUMENT);
+
+    mBleEndPoint = endPoint;
+
+    // Manually trigger the OnConnectComplete callback.
+    OnEndPointConnectComplete(endPoint, err);
+
+exit:
+    return err;
+}
+
+CHIP_ERROR BLEBase::SendMessage(const PacketHeader & header, const Transport::PeerAddress & address,
+                                System::PacketBufferHandle msgBuf)
+{
+    ReturnErrorCodeIf(address.GetTransportType() != Type::kBle, CHIP_ERROR_INVALID_ARGUMENT);
+    ReturnErrorCodeIf(mState == State::kNotReady, CHIP_ERROR_INCORRECT_STATE);
+    // ReturnErrorCodeIf(mBleEndPoint == nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    ReturnErrorOnFailure(header.EncodeBeforeData(msgBuf));
+
+    if (mState == State::kConnected)
+    {
+        ReturnErrorOnFailure(mBleEndPoint->Send(std::move(msgBuf)));
+    }
+    else
+    {
+        ReturnErrorOnFailure(SendAfterConnect(std::move(msgBuf)));
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR BLEBase::SendAfterConnect(System::PacketBufferHandle msg)
+{
+    // This will initiate a connection to the specified peer
+    CHIP_ERROR err = CHIP_ERROR_NO_MEMORY;
+
+    // Iterate through the ENTIRE array. If a pending packet for
+    // the address already exists, this means a connection is pending and
+    // does NOT need to be re-established.
+    for (size_t i = 0; i < mPendingPacketsSize; i++)
+    {
+        if (mPendingPackets[i].IsNull())
+        {
+            ChipLogDetail(Inet, "Message appended to send queue");
+            mPendingPackets[i] = std::move(msg);
+            err                = CHIP_NO_ERROR;
+            break;
+        }
+    }
+
+    return err;
+}
+
+void BLEBase::OnBleConnectionComplete(Ble::BLEEndPoint * endpoint)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    ChipLogDetail(Inet, "BleConnectionComplete");
+
+    mBleEndPoint = endpoint;
+
+    // Initiate CHIP over BLE protocol connection.
+    err = mBleEndPoint->StartConnect();
+    SuccessOrExit(err);
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        if (mBleEndPoint)
+        {
+            mBleEndPoint->Close();
+            mBleEndPoint = nullptr;
+        }
+        ChipLogError(Ble, "Failed to setup ble endpoint: %s", ErrorStr(err));
+    }
+}
+
+void BLEBase::OnBleConnectionError(BLE_ERROR err)
+{
+    ChipLogDetail(Inet, "BleConnection Error: %s", ErrorStr(err));
```

Comment:
I replaced all Ble to Inet in this file instead, this file does nothing about actual BLE operations except BleEndPoint, and our packet is not passed to BLE stack when reached this file actually.
Since we make Ble a transport as UDP & TCP, `Inet` tag is better than `Ble` for me.

Reply:
Im' not going to argue much about that. But at least I'm glad that we are consistent.